src/platform_impl/macos/util/mod.rs

```diff
@@ -118,8 +114,7 @@ pub unsafe fn app_name() -> Option<id> {
 }
 
 pub unsafe fn superclass<'a>(this: &'a Object) -> &'a Class {
-    let superclass: id = msg_send![this, superclass];
-    &*(superclass as *const _)
+    msg_send![this, superclass]
 }
```

Comment:
`objc::runtime::Class` is equivalent to `objc_class`, not the typedef `Class` you're referring to. Yes, confusing.

Anyhow, I found these issues by using a modified version of [`objc`'s `"verify_message"` feature](https://docs.rs/objc/0.2.7/objc/#message-type-verification); I rewrote this because the `msg_send!` was returning `id`, which does not have the same encoding as `*const Class`.

So this is not a soundness-changing issue (because we convert the `id` to `*const Class` afterwards), though I think the code is clearer this way IMO.

We can also do:
```rust
let superclass: *const Class = msg_send![this, superclass];
&*superclass
```

Reply:
Yes, this one seems much better!